error.go

```diff
@@ -186,13 +186,13 @@ func IsTestReleaseTimeout(err error) bool {
 	return microerror.Cause(err) == testReleaseTimeoutError
 }
 
-var tillerImageNotFoundError = &microerror.Error{
-	Kind: "tillerImageNotFoundError",
+var tillerImageInvalidError = &microerror.Error{
+	Kind: "tillerImageInvalidError",
```

Comment:
Renamed the error to make it more generic. It is returned if we can't get the image or if the format is invalid.

Reply:
I don't think creating very specific errors just in case makes sense in general. We don't track where we handle them and then are afraid to change them. I'd go with `executionFailedError` with a nice description everywhere and extract a specific error if it needs to be handled somewhere.

Reply:
My comment applies also to `tillerImageInvalidError` and other newly intorduced errors.

Reply:
OK cool that makes sense. I see the problem with making it hard to change errors later. 

I'm going to keep `tillerOutdatedError` because I'll need to test for that.